pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,450 @@
+                           min: orig_size,
+                           max: max_size,
+                           allow_infinite,
+                           round: round_size,
+                       })
+        ];
+    } else {
+        grow_size = block.Size;
+    }
+
+    let recovered_passphrase;
+    let passphrase_fields = [];
+    if (block && block.IdType == "crypto_LUKS" && block.IdVersion == 2)
+        passphrase_fields = existing_passphrase_fields(_("Resizing an encrypted filesystem requires unlocking the disk. Please provide a current disk passphrase."));
+
+    if (!usage.Teardown && size_fields.length + passphrase_fields.length === 0) {
+        return lvol_or_part_and_fsys_resize(client, lvol_or_part, grow_size, info.grow_needs_unmount, null);
+    }
+
+    const dlg = dialog_open({
+        Title: title,
+        Teardown: TeardownMessage(usage),
+        Fields: size_fields.concat(passphrase_fields),
+        Action: {
+            Title: _("Grow"),
+            action: function (vals) {
+                return teardown_active_usage(client, usage)
+                        .then(function () {
+                            return (lvol_or_part_and_fsys_resize(client, lvol_or_part,
+                                                                 to_fit ? grow_size : vals.size,
+                                                                 info.grow_needs_unmount,
+                                                                 vals.passphrase || recovered_passphrase)
+                                    .catch(request_passphrase_on_error_handler(dlg, vals, recovered_passphrase, block)));
+                        });
+            }
+        },
+        Inits: [
+            init_active_usage_processes(client, usage),
+            passphrase_fields.length
+                ? init_existing_passphrase(block, false, pp => { recovered_passphrase = pp })
+                : null
+        ]
+    });
+}
+
+export function shrink_dialog(client, lvol_or_part, info, to_fit) {
+    let title, block, name, orig_size, round_size;
+
+    if (lvol_or_part.iface == "org.freedesktop.UDisks2.LogicalVolume") {
+        const vgroup = client.vgroups[lvol_or_part.VolumeGroup];
+
+        title = _("Shrink logical volume");
+        block = client.lvols_block[lvol_or_part.path];
+        name = lvol_or_part.Name;
+        orig_size = lvol_or_part.Size;
+        round_size = vgroup.ExtentSize;
+    } else {
+        title = _("Shrink partition");
+        block = client.blocks[lvol_or_part.path];
+        name = block_name(block);
+        orig_size = lvol_or_part.Size;
+        round_size = 1024 * 1024;
+    }
+
+    const usage = get_active_usage(client, block && !to_fit && info.shrink_needs_unmount ? block.path : null,
+                                   _("shrink"));
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), name),
+            Body: BlockingMessage(usage)
+        });
+        return;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230925-111150-44391ef2-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)